providers/dns/brandit/internal/client.go

```diff
@@ -0,0 +1,244 @@
+	if err != nil {
+		return nil, fmt.Errorf("list records: %w", err)
+	}
+
+	result, err := c.AddRecord(domainName, account.Response.Registrar[0], fmt.Sprint(recordID.Response.Total[0]), record)
+	if err != nil {
+		return nil, fmt.Errorf("add record: %w", err)
+	}
+
+	return result, nil
+}
+
+func (c *Client) CleanUpRecord(domainName, dnsRecord string) (*DeleteRecord, error) {
+	// find the account associated with the domain
+	account, err := c.StatusDomain(domainName)
+	if err != nil {
+		return nil, fmt.Errorf("status domain: %w", err)
+	}
+
+	records, err := c.ListRecords(account.Response.Registrar[0], domainName)
+	if err != nil {
+		return nil, fmt.Errorf("list records: %w", err)
+	}
+	var recordID int
+	for i, r := range records.Response.Rr {
+		if r == dnsRecord {
+			recordID = i
+		}
+	}
+	result, err := c.DeleteRecord(domainName, account.Response.Registrar[0], dnsRecord, fmt.Sprint(recordID))
+	if err != nil {
+		return nil, fmt.Errorf("delete record: %w", err)
+	}
+
+	return result, nil
+}
+
+// DeleteRecord deletes a DNS record.
+// https://portal.brandit.com/apidocv3#deleteDNSRR
+func (c *Client) DeleteRecord(domainName, account, dnsRecord, recordID string) (*DeleteRecord, error) {
+	// Create a new query
+	query := url.Values{}
+	query.Add("command", "deleteDNSRR")
+	query.Add("account", account)
+	query.Add("dnszone", domainName)
+	query.Add("rrdata", dnsRecord)
+	query.Add("key", recordID)
+
+	result := &DeleteRecord{}
+
+	err := c.do(query, result)
+	if err != nil {
+		return nil, fmt.Errorf("do: %w", err)
+	}
+
+	return result, nil
+}
+
+// StatusDomain returns the status of a domain and account associated with it.
+// https://portal.brandit.com/apidocv3#statusDomain
+func (c *Client) StatusDomain(domain string) (*StatusDomain, error) {
+	// Create a new query
+	query := url.Values{}
+
+	query.Add("command", "statusDomain")
+	query.Add("domain", domain)
+
+	result := &StatusDomain{}
+
+	err := c.do(query, result)
+	if err != nil {
+		return nil, fmt.Errorf("do: %w", err)
+	}
+
+	return result, nil
+}
+
+func (c *Client) do(query url.Values, result any) error {
+	// Add signature
+	v, err := sign(c.apiUsername, c.apiKey, query)
+	if err != nil {
+		return fmt.Errorf("signature: %w", err)
+	}
+
+	resp, err := c.HTTPClient.PostForm(c.BaseURL, v)
+	if err != nil {
+		return err
+	}
+
+	defer func() { _ = resp.Body.Close() }()
+
+	all, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("read response body: %w", err)
+	}
+
+	//  Unmarshal the error response, because the API returns a 200 OK even if there is an error.
+	var Err Errors
+	err = json.Unmarshal(all, &Err)
+	if err != nil {
+		return fmt.Errorf("unmarshal response body: %w", err)
+	}
+	if Err.Code > 299 {
+		return fmt.Errorf("response code: %d, message: %s", Err.Code, Err.Error)
+	}
+
+	err = json.Unmarshal(all, result)
+	if err != nil {
+		return fmt.Errorf("unmarshal response body: %w", err)
+	}
```

Comment:
```suggestion
	raw, err := io.ReadAll(resp.Body)
	if err != nil {
		return fmt.Errorf("read response body: %w", err)
	}

	//  Unmarshal the error response, because the API returns a 200 OK even if there is an error.
	var apiError APIError
	err = json.Unmarshal(raw, &apiError)
	if err != nil {
		return fmt.Errorf("unmarshal error response: %w %s", err, string(raw))
	}

	if apiError.Code > 299 || apiError.Status != "success" {
		return apiError
	}

	err = json.Unmarshal(raw, result)
	if err != nil {
		return fmt.Errorf("unmarshal response body: %w %s", err, string(raw))
	}
```